detectors/CheckDetector.cpp

```diff
@@ -0,0 +1,41 @@
+#include "CheckDetector.h"
+
+bool CheckDetector::playerInCheck(Color player, Board& board) {
+
+    std::vector<Piece*> enemyPieces;
+    Piece *attackedKind = nullptr;
+
+    for(int i = 1; i < 9; i ++){
+        for(char j = 'A'; j < 'I'; j++){
+            Piece *p = board.getPiece({ j, i });
+            if(p != nullptr && p->player != player) {
+                enemyPieces.push_back(p);
+            }
+            if(p != nullptr && p->getCLIToken() == 'K' && p->player == player){
+                attackedKind = p;
+            }
+        }
+    }
+
+    bool res = false;
+
+    if(attackedKind != nullptr) {
```

Comment:
Why would there ever not be an attacked king? Shouldn't this throw an error since it might cause other problems?

Reply:
In theory there should always bean attacked king but really scared of nullptr haha. Can throw error if the king is null.